historical/common/dynamodb.py

```diff
@@ -84,41 +109,165 @@ def delete_differ_record(old_image, durable_model, region_attr):
     log.debug('[+] Adding deletion marker.')
 
 
-def deserialize_current_to_durable(record, current_model, durable_model):
+def get_full_current_object(record, current_model):
+    """
+    Utility method to fetch items from the Current table if they are too big for SNS/SQS.
+
+    :param record:
+    :param current_model:
+    :return:
+    """
+    arn = record['dynamodb']['Keys']['arn']['S']
+    log.debug('[-->] Item with ARN: {} was too big for SNS -- fetching it from the Current table...'.format(arn))
+    item = list(current_model.query(arn))
+
+    # If for whatever reason, the item *cannot* be found, then this record should be skipped over.
+    # This will happen if this event came in and got processed right after the item was deleted
+    # from the Current table. If so, then do nothing -- the deletion event will be processed later.
+    if not item:
+        return None
+
+    # We need to place the real configuration data into the record so it can be deserialized into
+    # the current model correctly:
+    record['dynamodb']['NewImage']['configuration'] = item[0]._serialize()['attributes']['configuration']
+
+    return record
+
+
+def get_full_durable_object(record, durable_model):
+    """
+    Utility method to fetch items from the Durable table if they are too big for SNS/SQS.
+
+    :param record:
+    :param durable_model:
+    :return:
+    """
+    arn = record['dynamodb']['Keys']['arn']['S']
+    event_time = record['dynamodb']['NewImage']['eventTime']['S']
+    log.debug('[-->] Item with ARN: {} was too big for SNS -- fetching it from the Durable table...'.format(arn))
+    item = list(durable_model.query(arn, durable_model.eventTime == event_time))
+
+    # It is not clear if this would ever be the case... that being said... let's log for it because we would need
+    # to examine why this happens (if it does):
+    if not item:
```

Comment:
If this happens won't it cause a failure condition (exception) that would be handled by the DLQ instead of merely logging it?